src/oca_pre_commit_hooks/global_parser.py

```diff
@@ -0,0 +1,70 @@
+import argparse
+import configparser
+from os import getcwd
+from os.path import exists, expanduser, join
+
+CONFIG_NAME = ".oca_hooks.cfg"
+MSG_CTRL = "MESSAGES_CONTROL"
+
+
+def parse_disable(value):
+    return set(value.split(","))
+
+
+class GlobalParser(argparse.ArgumentParser):
+    def __init__(self):
+        super().__init__()
+        self.add_argument(
+            "--no-verbose",
+            action="store_true",
+            default=False,
+            help="If enabled so disable verbose mode.",
+        )
+        self.add_argument(
+            "--no-exit",
+            action="store_true",
+            default=False,
+            help="If enabled so it will not call exit.",
+        )
+        self.add_argument(
+            "--disable",
+            "-d",
+            type=parse_disable,
+            default=set(),
+            help="Disable the checker with the given 'check-name', separated by commas.",
+        )
+        self.add_argument(
+            "--enable",
+            "-e",
+            type=parse_disable,
+            default=set(),
+            help=(
+                "Enable the checker with the given 'check-name', separated by commas. "
+                "Default: All checks are enabled by default"
+            ),
+        )
+        self.add_argument("--config", "-c", type=str, help="Path to a configuration file")
+
+    def parse_args(self, args=None, namespace=None):
+        res = super().parse_args(args)
+
+        if not res.config and exists(join(getcwd(), CONFIG_NAME)):
+            res.config = join(getcwd(), CONFIG_NAME)
+        elif exists(join(expanduser("~"), CONFIG_NAME)):
+            res.config = join(expanduser("~"), CONFIG_NAME)
```

Comment:
Could you move this piece of code to top_path method, please?
 - https://github.com/OCA/odoo-pre-commit-hooks/blob/7310b9f7a4f3ae1cf28092728e2b911198d283e4/src/oca_pre_commit_hooks/utils.py#L109

I mean, return HOME instead of DRIVE if it is not a `.git repo`

The precedence should be

cwd or git_top or home